File: app/src/main/java/com/example/orderingapp/Helper/ManagmentCart.java
package com.example.orderingapp.Helper;

import android.content.Context;
import android.widget.Toast;

import com.example.orderingapp.Domain.ItemsModel;

import java.util.ArrayList;

public class ManagmentCart {

    private final Context context;
    private final TinyDB tinyDB;

    public ManagmentCart(Context context) {
        this.context = context;
        this.tinyDB = new TinyDB(context);
    }

    public ArrayList<ItemsModel> getListCart() {
        ArrayList<ItemsModel> list = tinyDB.getListObject("CartList", ItemsModel.class);
        if (list == null) {
            list = new ArrayList<>();
        }
        return list;
    }

    public void insertItems(ItemsModel item) {
        ArrayList<ItemsModel> listItem = getListCart();
        int index = -1;
        for (int i = 0; i < listItem.size(); i++) {
            if (listItem.get(i).getTitle().equals(item.getTitle())) {
                index = i;
                break;
            }
        }

        if (index >= 0) {
            listItem.get(index).setNumberIncart(item.getNumberIncart());
        } else {
            listItem.add(item);
        }

        tinyDB.putListObject("CartList", listItem);
        Toast.makeText(context, "Added to your Cart", Toast.LENGTH_SHORT).show();
    }

    public void insertItemsList(ArrayList<ItemsModel> items) {
        tinyDB.putListObject("CartList", items);
    }

    public void plusItem(ArrayList<ItemsModel> listItems, int position, ChangeNumberItemsListener listener) {
        if (position >= 0 && position < listItems.size()) {
            ItemsModel item = listItems.get(position);
            item.setNumberIncart(item.getNumberIncart() + 1);
            tinyDB.putListObject("CartList", listItems);
            listener.onChanged();
        }
    }

    public void minusItem(ArrayList<ItemsModel> listItems, int position, ChangeNumberItemsListener listener) {
        if (position >= 0 && position < listItems.size()) {
            ItemsModel item = listItems.get(position);
            if (item.getNumberIncart() <= 1) {
                listItems.remove(position);
            } else {
                item.setNumberIncart(item.getNumberIncart() - 1);
            }
            tinyDB.putListObject("CartList", listItems);
            listener.onChanged();
        }
    }

    public void removeItem(ArrayList<ItemsModel> listItems, int position, ChangeNumberItemsListener listener) {
        if (position >= 0 && position < listItems.size()) {
            listItems.remove(position);
            tinyDB.putListObject("CartList", listItems);
            listener.onChanged();
        }
    }

    public double getTotalFee() {
        ArrayList<ItemsModel> listItem = getListCart();
        double fee = 0.0;
        for (ItemsModel item : listItem) {
            fee += item.getPrice() * item.getNumberIncart();
        }
        return fee;
    }
}

File: app/src/main/java/com/example/orderingapp/Helper/ManagmentCart.kt
package com.example.orderingapp.Activity.Helper

import android.content.Context
import android.widget.Toast
import com.example.orderingapp.Domain.ItemsModel
import com.example.orderingapp.Helper.TinyDB

class ManagmentCart(private val context: Context) {

    private val tinyDB = TinyDB(context)

    fun getListCart(): ArrayList<ItemsModel> {
        return tinyDB.getListObject("CartList", ItemsModel::class.java) ?: arrayListOf()
    }

    // Insert new item or update quantity to exact number in item
    fun insertItems(item: ItemsModel) {
        val listItem = getListCart()
        val index = listItem.indexOfFirst { it.title == item.title }

        if (index >= 0) {
            // Update quantity to the value inside item
            listItem[index].numberIncart = item.numberIncart
        } else {
            listItem.add(item)
        }

        tinyDB.putListObject("CartList", listItem)
        Toast.makeText(context, "Added to your Cart", Toast.LENGTH_SHORT).show()
    }

    // Save entire list at once
    fun insertItemsList(items: ArrayList<ItemsModel>) {
        tinyDB.putListObject("CartList", items)
    }

    // Increase quantity of item at position by 1
    fun plusItem(listItems: ArrayList<ItemsModel>, position: Int, listener: ChangeNumberItemsListener) {
        val item = listItems.getOrNull(position)
        item?.let {
            it.numberIncart++
            tinyDB.putListObject("CartList", listItems)
            listener.onChanged()
        }
    }

    // Decrease quantity or remove item if quantity is 1
    fun minusItem(listItems: ArrayList<ItemsModel>, position: Int, listener: ChangeNumberItemsListener) {
        val item = listItems.getOrNull(position)
        if (item != null) {
            if (item.numberIncart <= 1) {
                listItems.removeAt(position)
            } else {
                item.numberIncart--
            }
            tinyDB.putListObject("CartList", listItems)
            listener.onChanged()
        }
    }

    // Remove item completely by position
    fun removeItem(listItems: ArrayList<ItemsModel>, position: Int, listener: ChangeNumberItemsListener) {
        listItems.getOrNull(position)?.let {
            listItems.removeAt(position)
            tinyDB.putListObject("CartList", listItems)
            listener.onChanged()
        }
    }

    fun getTotalFee(): Double {
        val listItem = getListCart()
        var fee = 0.0
        for (item in listItem) {
            fee += item.price * item.numberIncart
        }
        return fee
    }
}
